### DISCUSSION TITLE
Expand home directory for FilePath with relative path

### DISCUSSION THREAD

**Discussion by ghost** [pydantic/pydantic#2807] [Feature Request]
# Feature Request

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
python -c "import pydantic.utils; print(pydantic.utils.version_info())"
             pydantic version: 1.8
            pydantic compiled: True
                 install path: /Users/james.tran/test/.venv/lib/python3.7/site-packages/pydantic
               python version: 3.7.9 (default, Apr 20 2021, 15:16:28)  [Clang 12.0.0 (clang-1200.0.32.29)]
                     platform: Darwin-20.4.0-x86_64-i386-64bit
     optional deps. installed: ['typing-extensions']
```

Would it be ok to allow expanding the `~` in the path for `FilePath`? This could be similar to the fix for https://github.com/samuelcolvin/pydantic/issues/1368

This snippet currently fails validation even though the file `~/dummy.txt` exists.
```bash
touch ~/dummy.txt
```

```python
from pydantic import BaseModel, FilePath

class Config(BaseModel):
    test_file_path: FilePath

config = Config(test_file_path="~/dummy.txt")
```

**Comment by PrettyWood** [comment]
Hi @jtran-xero
I reckon it would make sense to add the same logic as https://github.com/samuelcolvin/pydantic/pull/1804 indeed for convenience and consistence

**Reply by ghost** [reply]
@PrettyWood Thanks for the quick reply!

Sorry, I'm not very familiar with this workflow. Should I create a new issue that references this thread?
